src/commands/mod.rs

```diff
@@ -586,3 +586,54 @@ impl Formatter for PlaylistItems {
         )
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use tempfile::tempdir;
+    use std::io::Read;
+
+    #[test]
+    fn test_save_to_file_should_save_yaml_correctly() {
+        let dir = tempdir().unwrap();
+        let file_path = dir.path().join("test.yaml");
+
+        let manifest = EdgeAppManifest {
+            app_id: "test_app".to_string(),
+            settings: vec![Setting {
+                title: "username".to_string(),
+                type_: "string".to_string(),
+                default_value: "stranger".to_string(),
+                optional: true,
+                help_text: "An example of a setting that is used in index.html".to_string(),
+            }],
+            ..Default::default()
+        };
+
+        EdgeAppManifest::save_to_file(&manifest, &file_path).unwrap();
+
+        let mut file = File::open(&file_path).unwrap();
+        let mut contents = String::new();
+        file.read_to_string(&mut contents).unwrap();
+
+        let expected_contents = r#"---
+app_id: test_app
+user_version: ''
+description: ''
+icon: ''
+author: ''
+homepage_url: ''
+settings:
+  username:
+    type: string
+    default_value: stranger
+    title: username
+    optional: true
+    help_text: An example of a setting that is used in index.html
+"#;
+
+        assert_eq!(contents, expected_contents);
+
+        dir.close().unwrap();
```

Comment:
why do you need to call close manually?

Reply:
I think that it's a good practice and makes tests more robust. 

I've seen it it at leas one other test - should I delete it?